modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/portlet/action/GetDataMVCResourceCommand.java

```diff
@@ -211,6 +212,70 @@ private String _getClassTypeName(HttpServletRequest httpServletRequest) {
 		return ParamUtil.getString(httpServletRequest, "classTypeName");
 	}
 
+	private JSONObject _getEndpointsJSONObject(
+		AnalyticsReportsInfoItem<Object> analyticsReportsInfoItem,
+		String canonicalURL, Locale locale, Object object,
+		ResourceResponse resourceResponse) {
+
+		JSONObject jsonObject = JSONFactoryUtil.createJSONObject();
+
+		List<AnalyticsReportsInfoItem.Action> actions =
+			analyticsReportsInfoItem.getActions();
+
+		if (actions.contains(
+				AnalyticsReportsInfoItem.Action.HISTORICAL_READS)) {
+
+			jsonObject.put(
+				"analyticsReportsHistoricalReadsURL",
+				String.valueOf(
```

Comment:
This piece of code it's almost duplicated, only the MVC command name is different from each of them. Maybe here we can extract some code? What do you think?

Reply:
Maybe, as we dig into the actions a little bit more, maybe new patterns could be applied, for now, as we don't exactly where are we going, let keep it as simple as possible.